import "libnutrition/proto/service/single_food_messages.proto";
import "libnutrition/proto/service/composite_food_messages.proto";
import "libnutrition/proto/service/template_messages.proto";
import "libnutrition/proto/service/meal_messages.proto";
import "libnutrition/proto/service/nutrient_messages.proto";
import "libnutrition/proto/service/unit_messages.proto";
import "libnutrition/proto/service/specialized_unit_messages.proto";
import "libnutrition/proto/service/group_messages.proto";
import "libnutrition/proto/service/user_messages.proto";
import "libnutrition/proto/service/error.proto";

message DataLoadRequest 
{
	optional SingleFoodLoadRequest singleFoodLoadRequest = 1;
	
	optional CompositeFoodLoadRequest compositeFoodLoadRequest = 2;
	
	optional TemplateLoadRequest templateLoadRequest = 3;
	
	optional MealLoadRequest mealLoadRequest = 4;	
	
	optional NutrientLoadRequest nutrientLoadRequest = 5;
	
	optional UnitLoadRequest unitLoadRequest = 6;
		
	optional SpecializedUnitLoadRequest specializedUnitLoadRequest = 7;

	optional GroupLoadRequest groupLoadRequest = 8;
	
	optional UserLoadRequest userLoadRequest = 9;	
}

message DataLoadResponse
{
	optional SingleFoodLoadResponse singleFoodLoadResponse = 1;
	
	optional CompositeFoodLoadResponse compositeFoodLoadResponse = 2;
	
	optional TemplateLoadResponse templateLoadResponse = 3;
	
	optional MealLoadResponse mealLoadResponse = 4;	
	
	optional NutrientLoadResponse nutrientLoadResponse = 5;
	
	optional UnitLoadResponse unitLoadResponse = 6;
		
	optional SpecializedUnitLoadResponse specializedUnitLoadResponse = 7;

	optional GroupLoadResponse groupLoadResponse = 8;
	
	optional UserLoadResponse userLoadResponse = 9;
	
	optional Error error = 100;
}

message DataStoreRequest 
{
	optional SingleFoodStoreRequest singleFoodStoreRequest = 1;
	
	optional CompositeFoodStoreRequest compositeFoodStoreRequest = 2;
	
	optional TemplateStoreRequest templateStoreRequest = 3;
	
	optional MealStoreRequest mealStoreRequest = 4;	
	
	optional UserStoreRequest userStoreRequest = 9;	
}

message DataStoreResponse
{
	optional SingleFoodStoreResponse singleFoodStoreResponse = 1;
	
	optional CompositeFoodStoreResponse compositeFoodStoreResponse = 2;
	
	optional TemplateStoreResponse templateStoreResponse = 3;
	
	optional MealStoreResponse mealStoreResponse = 4;
	
	optional UserStoreRequest userStoreResponse = 9;	
	
	optional Error error = 100;
}

message DataDeleteRequest 
{
	optional SingleFoodDeleteRequest singleFoodDeleteRequest = 1;
	
	optional CompositeFoodDeleteRequest compositeFoodDeleteRequest = 2;
	
	optional TemplateDeleteRequest templateDeleteRequest = 3;
	
	optional MealDeleteRequest mealDeleteRequest = 4;		
}

message DataDeleteResponse
{
	optional SingleFoodDeleteResponse singleFoodDeleteResponse = 1;
	
	optional CompositeFoodDeleteResponse compositeFoodDeleteResponse = 2;
	
	optional TemplateDeleteResponse templateDeleteResponse = 3;
	
	optional MealDeleteResponse mealDeleteResponse = 4;	
	
	optional Error error = 100;
}